src/main/python/RLBotFramework/utils/rendering/rendering_manager.py

```diff
@@ -0,0 +1,199 @@
+        """
+        return self
+
+    def draw_line_2d_3d(self, x, y, vec, color):
+        """
+        :param x: represents the start of the line in the x coordinate
+        :param y: represents the start of the line in the y coordinate
+        :param vec: This should be a tutple or an array
+        :param color: Should be created by calling create_color
+        :return: self for chaining
+        """
+        return self
+
+    def draw_rect_2d(self, x, y, width, height,  filled, color):
+        """
+        :param x: represents the left coordinate of the rectangle
+        :param y: represents the top coordinate of the rectangle
+        :param width: How wide the rectangle should be
+        :param height: How tall the rectangle should be
+        :param filled: True if the rectangle should be filled in, false if it should be an outline
+        :param color: Should be created by calling create_color
+        :return: self for chaining
+        """
+        return self
+
+    def draw_rect_3d(self, vec, width, height,  filled, color):
+        """
+        :param vec: This should be a tuple or an array
+        :param width: How wide the rectangle should be
+        :param height: How tall the rectangle should be
+        :param filled: True if the rectangle should be filled in, false if it should be an outline
+        :param color: Should be created by calling create_color
+        :return: self for chaining
+        """
+        return self
+
+    def draw_string_2d(self, x, y, scale_x, scale_y, text, color):
+        """
+        :param x: represents the left coordinate of the rectangle
+        :param y: represents the top coordinate of the rectangle
+        :param scale_x: How much to scale the text by in the x direction
+        :param scale_y: How much to scale the text by in the y direction
+        :param text: The string
+        :param color: Should be created by calling create_color
+        :return: self for chaining
+        """
+        return self
+
+    def draw_string_3d(self, vec, scale_x, scale_y, text, color):
+        """
+        :param vec: This should be a tuple or an array
+        :param scale_x: How much to scale the text by in the x direction
+        :param scale_y: How much to scale the text by in the y direction
+        :param text: The string
+        :param color: Should be created by calling create_color
+        :return: self for chaining
+        """
+        return self
+
+    def create_color(self, alpha, red, green, blue):
+        color = Color()
+        color.a = alpha
+        color.r = red
+        color.g = green
+        color.b = blue
+        return color
+
+    def black(self):
+        return self.create_color(255, 0, 0, 0)
+
+    def white(self):
+        return self.create_color(255, 255, 255, 255)
+
+    def get_render_functions(self):
+        """
+        Gets all the raw render functions but without giving access to any internal logic or the dll

```

Comment:
Why not just use the `__` prefix for private stuff? That should be enough to prevent accidental misuse. Is there some benefit to this dynamic function technique that I'm missing?

Reply:
A bot could also construct an additional instance of the game interface without any help from the rendering system. Malicious tournament players have a ton of options, and we're pretty much on the honor system given the current architecture.

Reply:
Done